### ISSUE TITLE
Support for Infallible error.

### ISSUE THREAD

**Issue by andy-thomason** [extendr/extendr#234]
This is a small issue to add `Infallible` conversion to our error type.

https://doc.rust-lang.org/std/convert/enum.Infallible.html

`Infallible` is the "never" error type which means that it will never be actually thrown.

However, we need to be able to convert this type to our Error so that we can use any "From"
as a "TryFrom".

```
impl From<Infallible> for Error {
    fn from(x: Infallible) -> Error {
       ...
    }
}
```

**Comment by yutannihilation** [comment]
I still don't understand the mechanism of this, but do we really have infallible conversion? I feel all `TryFrom<Robj>` can fail.

**Comment by clauswilke** [comment]
@yutannihilation I think the point is that if you have a `From` implementation for a particular conversion and you have `Infallible` then `TryFrom` will automatically be available for that same conversion. In this way, there can never be a situation where `From` is available by `TryFrom` is not, and therefore we can use `TryFrom` as the default (and get rid of the option in the extendr macro).

**Comment by andy-thomason** [comment]
Yes, `TryFrom` has a default implementation that takes a `From`.

When using the `?` operator, it is common to have `Infallible` as the error type
not just here but anywhere where an error will never occur.

**Comment by yutannihilation** [comment]
Sorry, I forgot to reply here. I got it, thanks.
